### ISSUE TITLE
It can't play black under Fritz 17

### ISSUE THREAD

**Issue by SzotsGabor** [redsalmon91/FoxSEE#3]
I have just encountered a problem. FoxSEE does not move under Fritz 17 when playing black. This was in a tournament, it played the first game with white without problems but it did not move in the second game playing black. I tried versions 4.0.1, 4.0.0 and 3.3.3.
There is no problem under Arena or Shredder.

**Comment by buildingwheels** [comment]
Thank you, Gabor. I will take a look.
My testing GUIs are Shredder 13, CuteChess, and Banksia. I have not yet used Fritz 17. It might have a different implementation of the UCI protocol.

Is it possible that Fritz 17 has kept a crash log of the engine?
Thanks.

**Comment by SzotsGabor** [comment]
Hi Han Zixiao,

I am not that familiar with Fritz so even if it stores a log file 
somewhere I don't know where. I am going to do some investigation about 
that and if I find something I'll let you know.

Gabor

------ Eredeti üzenet ------
Feladó: "Zixiao Han" <notifications@github.com>
Címzett: "redsalmon91/FoxSEE" <FoxSEE@noreply.github.com>
Másolat: "SzotsGabor" <szots.gabor324@gmail.com>; "Author" 
<author@noreply.github.com>
Elküldve: 2020.07.10. 14:40:13
Tárgy: Re: [redsalmon91/FoxSEE] It can't play black under Fritz 17 (#3)

**Comment by buildingwheels** [comment]
Much appreciated! @SzotsGabor I may get a copy of the Fritz 17 as well, after moving is completed and I get access to my win desktop again.

**Comment by SzotsGabor** [comment]
I could not find a way to log communication and was told at CCC that there is none. I haven't tried v5.0.0 yet.

**Comment by buildingwheels** [comment]
> I could not find a way to log communication and was told at CCC that there is none. I haven't tried v5.0.0 yet.

5.0.0 does not have any change in protocol.
I will get a copy of Frtiz myself later to find out what is different in their UCI implementation. I currently do not have access to my windows desktop.
Thank you Gabor.

**Comment by SzotsGabor** [comment]
No pressure. I am currently running it under Shredder where there is no 
problem.

------ Eredeti üzenet ------
Feladó: "Zixiao Han" <notifications@github.com>
Címzett: "redsalmon91/FoxSEE" <FoxSEE@noreply.github.com>
Másolat: "SzotsGabor" <szots.gabor324@gmail.com>; "Mention" 
<mention@noreply.github.com>
Elküldve: 2020.07.11. 12:15:28
Tárgy: Re: [redsalmon91/FoxSEE] It can't play black under Fritz 17 (#3)

**Comment by rwbc** [comment]
Well, Inbetween could log all communication for you.

You have already used it at least once for a more difficult case (java program) :)
Here again my detailed report about how to use it with screenshots.

http://talkchess.com/forum3/viewtopic.php?f=2&t=63226&p=707012&hilit=fritz+inbetween#p706985

> 
> No pressure. I am currently running it under Shredder where there is no problem.


> >I could not find a way to log communication and was told at CCC that >there is none.

**Comment by buildingwheels** [comment]
> Well, Inbetween could log all communication for you.
> 
> You have already used it at least once for a more difficult case (java program) :)
> Here again my detailed report about how to use it with screenshots.
> 
> http://talkchess.com/forum3/viewtopic.php?f=2&t=63226&p=707012&hilit=fritz+inbetween#p706985
> 
> > No pressure. I am currently running it under Shredder where there is no problem.
> 
> > > I could not find a way to log communication and was told at CCC that >there is none.

Appreciated. Was quite busy recently, just see this comment now.
Unfortunately, TalkChess has blocked all IPs from China, so I am not able to access the link that you have provided. Is it possible for me to ask for a copy of the thread?

**Comment by rwbc** [comment]
> 
> 
> > Well, Inbetween could log all communication for you.
> > You have already used it at least once for a more difficult case (java program) :)
> > Here again my detailed report about how to use it with screenshots.
> > http://talkchess.com/forum3/viewtopic.php?f=2&t=63226&p=707012&hilit=fritz+inbetween#p706985
> > > No pressure. I am currently running it under Shredder where there is no problem.
> > 
> > 
> > > > I could not find a way to log communication and was told at CCC that >there is none.
> 
> Appreciated. Was quite busy recently, just see this comment now.
> Unfortunately, TalkChess has blocked all IPs from China, so I am not able to access the link that you have provided. Is it possible for me to ask for a copy of the thread?

I hoped Gabor would read it first ;-)

Another solution would be to try Opera, it has a simple pseudo vpn mode, which is enough to get access to talkchess.
(Settings => Advanced => Use VPN ...)
I could still try it, if you think the Opera solution is too much work for you?

The original talkchess thread also contains screenshots showing it at work.
Note that it is for Windows only!

https://web.archive.org/web/20111116213422/http://home.online.no/~malin/sjakk/
Basically you download Inbetween from Odd Gunnar Malin and add appropriate lines into an ini file, e.g.
(you just rename the log name and put the real exe name into the commandline)
Debug :=3 means even a visible log window.
In the Fritz GUI you load the inbetween.exe, which I usually rename to sth similar than the target exe,
in the example below I would use FoxSEE_520.exe.

The options below can change / oppress output from both sides...

```
[InBetween]
Debug := 3
Log := FoxSEE_520.log
CommandLine := FoxSEE520.exe
 
[Client2Server]
; Set transalation of commands from the client here
; format word := translation
 
[Server2Client]
; Set transalation of commands from the server here
; format word := translation
```

**Comment by SzotsGabor** [comment]
> 
> 
> Well, Inbetween could log all communication for you.
> 
> You have already used it at least once for a more difficult case (java program) :)
> Here again my detailed report about how to use it with screenshots.
> 
> http://talkchess.com/forum3/viewtopic.php?f=2&t=63226&p=707012&hilit=fritz+inbetween#p706985
> 
> > No pressure. I am currently running it under Shredder where there is no problem.
> 
> > > I could not find a way to log communication and was told at CCC that >there is none.

Yes, I remember having used it with Carballo. However, I am not interested in InBetween or any other devices that require extra work any more. I am too lazy. Arena is capable of everything. I can live with not using java engines under Fritz.

**Comment by rwbc** [comment]
> 
> 
> > Well, Inbetween could log all communication for you.
> > You have already used it at least once for a more difficult case (java program) :)
> > Here again my detailed report about how to use it with screenshots.
> > http://talkchess.com/forum3/viewtopic.php?f=2&t=63226&p=707012&hilit=fritz+inbetween#p706985
> > > No pressure. I am currently running it under Shredder where there is no problem.
> > 
> > 
> > > > I could not find a way to log communication and was told at CCC that >there is none.
> 
> Yes, I remember having used it with Carballo. However, I am not interested in InBetween or any other devices that require extra work any more. I am too lazy. Arena is capable of everything. I can live with not using java engines under Fritz.

I don't understand you? Carballo was just an example for how to use it at that time...
Here we are in a thread about an **issue you have with FoxSEE** in one of **your GUIs** and I offered a solution for logging.

**Comment by SzotsGabor** [comment]
> 
> 
> > > Well, Inbetween could log all communication for you.
> > > You have already used it at least once for a more difficult case (java program) :)
> > > Here again my detailed report about how to use it with screenshots.
> > > http://talkchess.com/forum3/viewtopic.php?f=2&t=63226&p=707012&hilit=fritz+inbetween#p706985
> > > > No pressure. I am currently running it under Shredder where there is no problem.
> > > 
> > > 
> > > > > I could not find a way to log communication and was told at CCC that >there is none.
> > 
> > 
> > Yes, I remember having used it with Carballo. However, I am not interested in InBetween or any other devices that require extra work any more. I am too lazy. Arena is capable of everything. I can live with not using java engines under Fritz.
> 
> I don't understand you? Carballo was just an example for how to use it at that time...
> Here we are in a thread about an **issue you have with FoxSEE** in one of **your GUIs** and I offered a solution for logging.

Günther, I understand that you offered a solution to have logging under Fritz. What I meant was that I was not going to use InBetween (or any other adapter for that matter) for whatever purpose. And it is entirely out of comfort, I simply don't want to bother with adapters, even if they are easy to use.
My solution for the FoxSEE problem: I don't use it under Fritz.

BTW, if the problem occurs only with FoxSEE (which I think is the only engine with that behaviour) then a simple comparison of its UCI communication with that of any other UCI engine should reveal the reason.

**Comment by SzotsGabor** [comment]
Rasmas Althof suggested this:
"I think your problems with FoxSEE and Fritz 17 may be related to FoxSEE's incomplete UCI implementation. If I understand the code correctly, FoxSEE accepts either startpos plus movelist, or FEN but ignores the following movelist. If Fritz 17 handles it like Droidfish, i.e. handing over FEN plus movelist since the last irreversible move, then this wouldn't work. It would also explain why it works with some other GUIs. Maybe that would be a hint for FoxSEE's issue list."

**Comment by buildingwheels** [comment]
> Rasmas Althof suggested this:
> "I think your problems with FoxSEE and Fritz 17 may be related to FoxSEE's incomplete UCI implementation. If I understand the code correctly, FoxSEE accepts either startpos plus movelist, or FEN but ignores the following movelist. If Fritz 17 handles it like Droidfish, i.e. handing over FEN plus movelist since the last irreversible move, then this wouldn't work. It would also explain why it works with some other GUIs. Maybe that would be a hint for FoxSEE's issue list."

Thanks Gabor. Yes, I have received the same suggestion from Rasmus in Graham's live chatroom. The UCI parser change is the next one on my to-do list. I am currently working on a new version which potentially improves 150 elo over v7.8 (tested 2000 games in 40/2 tc). I will start the UCI parser change right after this functional change.

**Comment by buildingwheels** [comment]
This should be fixed now.
